### ISSUE TITLE
fit generator slow 

### ISSUE THREAD

**Issue by GeorgeXia1828** [fchollet/keras#3572]
I am trying to test the fit generator but found the speed is rather slow to around less 100 examples processing per second, I am wondering anything wrong in my code?

 `from **future** import print_function
import numpy as np
np.random.seed(1337)  # for reproducibility

from keras.datasets import mnist
from keras.models import Sequential
from keras.layers.core import Dense, Dropout, Activation
from keras.optimizers import SGD, Adam, RMSprop
from keras.utils import np_utils

nb_classes = 10
nb_epoch = 3

def gen_data(path="train.txt", batchsize=50):
    while 1:
        f = open(path)
        for nb_line, line in enumerate(f):
            batch_X = []
            batch_y = []
            line = line.strip("\n")
            line = line.split(",")
            line = [np.float32(i) for i in line]
            line_X = line[:-1]
            line_y = int(line[-1])
            batch_X.append(line_X)
            batch_y.append(line_y)
            if nb_line%50 == (batchsize-1):
                batch_X = np.array(batch_X)
                batch_y = np_utils.to_categorical(batch_y, 10)
                yield (batch_X, batch_y)
        f.close()

model = Sequential()
model.add(Dense(512, input_shape=(784,)))
model.add(Activation('relu'))
model.add(Dropout(0.2))
model.add(Dense(512))
model.add(Activation('relu'))
model.add(Dropout(0.2))
model.add(Dense(10))
model.add(Activation('softmax'))

model.summary()

model.compile(loss='categorical_crossentropy',
              optimizer=RMSprop(),
              metrics=['accuracy'])

history = model.fit_generator(gen_data(),samples_per_epoch=50000, nb_epoch=nb_epoch, nb_worker=5, pickle_safe=True)
model.save('george_mnist.h5')  `

the train.txt file format:
785 floats separate by comma, with the last as label.
